networking/eth2/src/integration-test/java/tech/pegasys/teku/networking/eth2/AttesterSlashingGossipIntegrationTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.teku.infrastructure.async.Waiter.waitFor;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.bls.BLSKeyGenerator;
+import tech.pegasys.teku.bls.BLSKeyPair;
+import tech.pegasys.teku.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.datastructures.util.DataStructureUtil;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.async.Waiter;
+import tech.pegasys.teku.networking.eth2.Eth2NetworkFactory.Eth2P2PNetworkBuilder;
+import tech.pegasys.teku.networking.eth2.gossip.GossipPublisher;
+import tech.pegasys.teku.networking.eth2.gossip.encoding.GossipEncoding;
+import tech.pegasys.teku.networking.eth2.gossip.topics.OperationProcessor;
+import tech.pegasys.teku.statetransition.validation.InternalValidationResult;
+
+public class AttesterSlashingGossipIntegrationTest {
+
+  private final List<BLSKeyPair> validatorKeys = BLSKeyGenerator.generateKeyPairs(3);
+  private final Eth2NetworkFactory networkFactory = new Eth2NetworkFactory();
+  private final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+
+  @AfterEach
+  public void tearDown() throws Exception {
+    networkFactory.stopAll();
+  }
+
+  @Test
+  public void shouldGossipToPeers() throws Exception {
```

Comment:
These tests mostly test if a gossiped slashing get included in the operation processor of a gossiped node. Should we not be testing if an operation added to the operation pool gets gossiped instead?

Reply:
Well, these are integration tests at the networking level.  So it's testing that given the right dependencies everything gets wired up correctly and the slashings actually flow through and get gossiped out externally.

It doesn't test that the publisher object is invoked correctly / wired up correctly at a higher level.  I could look into adding some tests around that, but its a little complicated because the wiring logic is buried in the BeaconChainController.

Reply:
Added a test to check that `OperationPool` subscriptions work.  So the only thing untested is the wiring logic between the operation pool and network which is pretty trivial.